crates/ruff_linter/src/rules/pycodestyle/rules/lambda_assignment.rs

```diff
@@ -267,14 +270,50 @@ fn replace_trailing_ellipsis_with_original_expr(
     )
     .unwrap_or(lambda.body.range());
 
-    let original_expr_in_source = checker.locator().slice(original_expr_range);
+    // This prevents the autofix of introducing a syntax error if the lambda's body is an
+    // expression spanned across multiple lines. To avoid the syntax error we preserve
+    // the parenthesis around the body.
+    let original_expr_in_source =
+        if is_expression_parenthesized(lambda.into(), checker.comment_ranges(), checker.source()) {
+            format!("({})", checker.locator().slice(original_expr_range))
+        } else {
+            checker.locator().slice(original_expr_range).to_string()
+        };
 
     let placeholder_ellipsis_start = generated.rfind("...").unwrap();
     let placeholder_ellipsis_end = placeholder_ellipsis_start + "...".len();
 
     generated.replace_range(
         placeholder_ellipsis_start..placeholder_ellipsis_end,
-        original_expr_in_source,
+        &original_expr_in_source,
     );
     generated
 }
+
+/// Returns `true` if the [`ExprRef`] is enclosed by parentheses in the source code.
+fn is_expression_parenthesized(
+    expr: ExprRef,
+    comment_ranges: &CommentRanges,
+    contents: &str,
+) -> bool {
+    // First test if there's a closing parentheses because it tends to be cheaper.
+    if matches!(
+        first_non_trivia_token(expr.end(), contents),
+        Some(SimpleToken {
+            kind: SimpleTokenKind::RParen,
+            ..
+        })
+    ) {
+        matches!(
+            BackwardsTokenizer::up_to(expr.start(), contents, comment_ranges)
+                .skip_trivia()
+                .next(),
+            Some(SimpleToken {
+                kind: SimpleTokenKind::LParen,
+                ..
+            })
+        )
+    } else {
+        false
+    }
+}
```

Comment:
Oh no, I pointed you to the wrong version it seems. There's one that can be reused. See `parenthesized_range`

Reply:
fixed